packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2276,9 +2276,8 @@ class LeaderLayer extends ContainerLayer {
   /// children.
   @override
   void applyTransform(Layer? child, Matrix4 transform) {
-    assert(_lastOffset != null);
-    if (_lastOffset != Offset.zero)
```

Comment:
It looks like _lastOffset is no longer useful in this class, maybe remove it?

Reply:
it looks like it was used in an assert in followerlayer. Maybe rename it to be debugLastOffset

Reply:
well probably just remove the assert, the assert is there to make sure we have lastoffset before calling  leader.applyTransform

Reply:
Yeah, it is still relevant for that assert. I'll rename it.

Reply:
This is interesting though, is it possible the leader._lastOffset is null when _establishTransform is called?

Reply:
Done.

Reply:
> This is interesting though, is it possible the leader._lastOffset is null when _establishTransform is called?

No, because _establishTransform can only be called when the LeaderLayer has a follower, which means `allwaysAddToScene` will be true, which means `addToScene` will have been called to set it.

Reply:
(or if _establishTransform is called when `leader._lastOffset` is null that means you are using it wrong as indicated by the assert message "LeaderLayer anchor must come before FollowerLayer in paint order, but the reverse was true.")